pkg/console/controllers/plugins/controller.go

```diff
@@ -0,0 +1,189 @@
+
+	// k8s
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+
+	// openshift
+	operatorv1 "github.com/openshift/api/operator/v1"
+	consoleinformersv1alpha1 "github.com/openshift/client-go/console/informers/externalversions/console/v1alpha1"
+	listerv1alpha1 "github.com/openshift/client-go/console/listers/console/v1alpha1"
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	// console-operator
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+	"github.com/openshift/console-operator/pkg/console/status"
+)
+
+const (
+	MigratedPluginsAnnotation = "console.openshift.io/migrated-plugins"
+)
+
+type PluginsMigrationController struct {
+	// configs
+	operatorClient       v1helpers.OperatorClient
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	// lister
+	consolePluginLister   listerv1alpha1.ConsolePluginLister
+	consolePluginInformer consoleinformersv1alpha1.ConsolePluginInformer
+}
+
+func NewPluginsMigrationController(
+	// operator
+	operatorClient v1helpers.OperatorClient,
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	// operatorConfigInformer operatorinformerv1.ConsoleInformer,
+	// plugins
+	consolePluginInformer consoleinformersv1alpha1.ConsolePluginInformer,
+	// events
+	recorder events.Recorder,
+) factory.Controller {
+
+	c := &PluginsMigrationController{
+		// configs
+		operatorClient:       operatorClient,
+		operatorConfigClient: operatorConfigClient,
+		// plugins
+		consolePluginLister: consolePluginInformer.Lister(),
+	}
+
+	return factory.New().
+		WithInformers(
+			consolePluginInformer.Informer(),
+		).WithSync(c.Sync).
+		ToController("PluginsMigrationController", recorder.WithComponentSuffix("plugins-migration-controller"))
+}
+
+func (c *PluginsMigrationController) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
+
+	operatorConfig, err := c.operatorConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorv1.Managed:
+		klog.V(4).Infof("console-operator is in a managed state: migrating available plugins")
+	case operatorv1.Unmanaged:
+		klog.V(4).Infof("console-operator is in an unmanaged state: skipping available plugins migration")
+		return nil
+	case operatorv1.Removed:
+		klog.V(4).Infof("console-operator is in a removed state: removing migrated plugins")
+		return c.removeMigratedPlugins(ctx, operatorConfig)
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	availablePluginsName, err := c.GetAvailablePluginsName()
```

Comment:
Honestly, I think this should be drive off the installed operator rather than the availability of the plugin. Anyone who can create ConsolePlugin resources can create one with the name `kubevrit`. We want to check instead that the CNV operator is installed in `openshift-cnv`.

Reply:
> We want to check instead that the CNV operator is installed in openshift-cnv.

so wondering if we want check if the CNV's CSV is installed in the `openshift-cnv` together with the ConsolePlugin's name?